api/stripe.go

```diff
@@ -35,55 +41,180 @@ func (a *API) handleWebhook(w http.ResponseWriter, r *http.Request) {
 	// Unmarshal the event data into an appropriate struct depending on its Type
 	switch event.Type {
 	case "customer.subscription.created":
-		customer, subscription, err := a.stripe.GetInfoFromEvent(*event)
-		if err != nil {
-			log.Errorf("stripe webhook: error getting info from event: %s\n", err.Error())
-			w.WriteHeader(http.StatusBadRequest)
-			return
-		}
-		address := subscription.Metadata["address"]
-		if len(address) == 0 {
-			log.Errorf("subscription %s does not contain an address in metadata", subscription.ID)
-			w.WriteHeader(http.StatusBadRequest)
-			return
-		}
-		org, _, err := a.db.Organization(address, false)
+		log.Infof("received stripe event Type: %s", event.Type)
+		stripeSubscriptionInfo, org, err := a.getSubscriptionOrgInfo(event)
 		if err != nil || org == nil {
 			log.Errorf("could not update subscription %s, a corresponding organization with address %s was not found.",
-				subscription.ID, address)
-			log.Errorf("please do manually for creator %s \n  Error:  %s", customer.Email, err.Error())
+				stripeSubscriptionInfo.ID, stripeSubscriptionInfo.OrganizationAddress)
+			log.Errorf("please do manually for creator %s \n  Error:  %s", stripeSubscriptionInfo.CustomerEmail, err.Error())
 			w.WriteHeader(http.StatusBadRequest)
 			return
 		}
```

Comment:
This code could be included in `getSubscriptionOrgInfo` helper, at least return an error when org is nil because the check is done every time that that function is executed.